From 431d5078573b6747ffbf1bdae6b783e912d67d18 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 5 Jul 2017 17:00:45 +0100
Subject: data: Configure D-BUS service file to use the full binary path

This patch ensures the D-BUS service file includes the full path for
the gnome-recipes binary, as specified at compile time.

D-BUS in a typical Linux distribution environment requires the use of
full paths, and doesn't permit a reliance on the environment to assist
launching or runtime.

This now ensures we can launch gnome-recipes using gapplication and
graphical launchers & menus, using D-BUS activation.

Bug GNOME: https://bugzilla.gnome.org/show_bug.cgi?id=784535

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 data/org.gnome.Recipes.service.in | 2 +-
 meson.build                       | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/data/org.gnome.Recipes.service.in b/data/org.gnome.Recipes.service.in
index d6281ae..6ceffba 100644
--- a/data/org.gnome.Recipes.service.in
+++ b/data/org.gnome.Recipes.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.gnome.Recipes
-Exec=gnome-recipes --gapplication-service
+Exec=@BINDIR@/gnome-recipes --gapplication-service
diff --git a/meson.build b/meson.build
index 8c14e25..5afc27b 100644
--- a/meson.build
+++ b/meson.build
@@ -106,10 +106,13 @@ conf.set_quoted('LOCALEDIR', join_paths([ get_option('prefix'),
                                           get_option('datadir'),
                                           'locale' ]))
 
-# Used while generating cuisine.css and the dbus service file
+# Used while generating cuisine.css
 conf.set_quoted('pkgdatadir', pkgdatadir)
 conf.set_quoted('bindir', bindir)
 
+# Must be unquoted for dbus service file
+conf.set('BINDIR', bindir)
+
 configure_file(output : 'config.h', configuration : conf)
 
 subdir('src')
-- 
cgit v0.12

